samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -267,22 +272,34 @@ public TestRunner addOutputStream(CollectionStream stream) {
     return this;
   }
 
+
   /**
    * Utility to run a test configured using TestRunner
+   *
+   * @param timeout time to wait for the high level application or low level task to finish. This timeout does not include
+   *                input stream initialization time or the assertion time over output streams. This timeout just accounts
+   *                for time that samza job takes run. When job times out a TimeoutException is thrown to the callee
    */
-  public void run() {
+  public void run(Duration timeout) throws SamzaException, TimeoutException {
```

Comment:
`SamzaException`  is a sub-type of `RuntimeException`, we don't have to add it in throws clause.

Reply:
Removed SamzaException here